js/components/entity-browser.js

```diff
@@ -0,0 +1,85 @@
+    'text!./cohort-definition-browser.html',
+    'components/Component',
+    'utils/Renderers',
+    'utils/CommonUtils',
+    'faceted-datatable',
+], function(ko, view, Component, renderers, commonUtils) {
+    class EntityBrowser extends Component {
+        constructor(params) {
+            super(params);
+            this.onSelect = params.onSelect;
+            this.multiChoice = params.multiChoice || false;
+            this.showModal = params.showModal;
+            this.scrollY = params.scrollY || '50vh';
+            this.scrollCollapse = params.scrollCollapse || false;
+            this.selectedDataIds =
+                this.multiChoice && params.selectedData
+                    ? (params.selectedData() || []).map(({ id }) => id)
+                    : [];
+            this.data = ko.observableArray([]);
+            this.isLoading = ko.observable(false);
+            this.importEnabled = ko.pureComputed(() => this.data().some(i => i.selected()));
+            this.filteredData = ko.observableArray([]);
+            this.buttons = !!this.multiChoice
+                ? [
+                      {
+                          text: 'Select All',
+                          action: () => this.toggleSelected(true),
+                          className: this.classes({ extra: 'btn btn-sm btn-success' }),
+                          init: this.removeClass('dt-button'),
+                      },
+                      {
+                          text: 'Deselect All',
+                          action: () => this.toggleSelected(false),
+                          className: this.classes({ extra: 'btn btn-sm btn-primary' }),
+                          init: this.removeClass('dt-button'),
+                      },
+                  ]
+                : null;
+
+            this.columns = !!this.multiChoice
+                ? [
+                    {
+                        data: 'selected',
+                        class: this.classes({extra: 'text-center'}),
+                        render: () => renderers.renderCheckbox('selected'),
+                        searchable: false,
+                        orderable: false,
+                    },
+                ] : [];
+
+            this.tableDom = "Bfiprt<'page-size'l>ip";
+            this.rowClick = this.rowClick.bind(this);
+            this.loadData();
+        }
+
+        rowClick(data) {
+            this.action(() => this.onSelect(data));
+        }
+
+        removeClass(className) {
+            return (dt, node, cfg) => node.removeClass(className);
+        }
+
+        action(callback) {
+            callback();
+        }
+
+        import() {
+            const selectedItems = this.data().filter(i => i.selected());
+            this.action(() => this.onSelect(selectedItems));
+        }
+
+        cancel() {
+            this.showModal(false);
+        }
+
+        toggleSelected(selected) {
+            commonUtils.selectAllFilteredItems(this.data, this.filteredData, selected);
```

Comment:
I'm still facing an issue with the following scenario that appears to be related to the fact that `filteredData` is not properly capturing the filter applied to the data table. Here are the steps to reproduce:

- Create a new cohort characterization
- Import cohort definitions to reveal the modal dialog 
- Use the search filter to filter down the list to a subset of cohort definitions.
- Click the "Select All" button to select the items that are part of the filtered set
- Click the "Import" button to import the selected cohorts into the design

When I put a breakpoint into this line, `filteredData` contains the full set of table entries and not the ones that are filtered based on my search criteria.

Reply:
Fixed